### PR TITLE
Update `wp-env` To 7.0.0

### PR DESCRIPTION
**Pull request by ObliviousHarmony**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This pull request updates `wp-env` to version 7.0.0. This new version includes a couple of upstream changes that we can utilize to simplify the bootstrapping of our E2E tests:

* https://github.com/WordPress/gutenberg/pull/50196: This will allow us to automatically run the E2E setup script without having to do so explicitly.
* https://github.com/WordPress/gutenberg/pull/49962: This fixes the permission errors that required us to manually set some of them to get things working.
* https://github.com/WordPress/gutenberg/pull/41179: This removes the need for quoting `run` arguments. There seems to be a bug with quotes being removed unless escaped, but, this is a nice little touch.

I've gone ahead and made some changes to the E2E bootstrapping to utilize these. We also _technically_ don't need the `update-wp-env.php` script at all, but, I wanted to keep that in place for this pull request. A subsequent one can remove that once the update is in place.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure E2E tests pass in CI.
2. Run `pnpm --filter=woocommerce env:dev` to update the environment with the changes that were made.
3. Test out the environment and make sure it still works as expected.

### CODE REVIEW

File: plugins/woocommerce/.wp-env.json [hunk 1]
```diff
@@ -10,13 +10,24 @@
...
-        "wp-cli.yml": ".\/tests\/wp-cli.yml"
+        "wp-cli.yml": "./tests/wp-cli.yml"
     },
+    "afterSetup": "./tests/e2e-pw/bin/test-env-setup.sh",
     "env": {
-        "development": [],
+        "development": {},
         "tests": {
             "port": 8086,
+            "plugins": [
```
Comment (lanej0) [inline plugins/woocommerce/.wp-env.json:20]: Such a nice change.  😍

File: plugins/woocommerce/package.json [hunk 2]
```diff
@@ -29,15 +29,15 @@
 		"docker:down": "pnpm exec wc-e2e docker:down",
 		"docker:ssh": "pnpm exec wc-e2e docker:ssh",
 		"docker:up": "pnpm exec wc-e2e docker:up",
-		"env:dev": "pnpm wp-env start",
+		"env:dev": "pnpm wp-env start --update",
```
Comment (lanej0) [inline plugins/woocommerce/package.json:32]: I'm noticing that `pnpm env:dev` and `pnpm env:start` do virtually the same thing.  Should we remove one?

**Review by lanej0 — APPROVED** [review]
Pulled the latest changes and everything is working as expected. 👍🏼

File: plugins/woocommerce/package.json (hunk 2 above)
Comment (ObliviousHarmony) [reply plugins/woocommerce/package.json:32]: I'm going to go through and clean up all of the scripts here once I've finished the series of PRs taking advantage of changes in `wp-env`.

**Pull request merged**